### PR TITLE
2.0-preview

### PR DESCRIPTION
**Pull request by cdata**
WIP

### CODE REVIEW

File: paper-ripple-behavior.html [hunk 1]
```diff
@@ -74,6 +74,7 @@
         var rippleContainer = this._rippleContainer || this.root;
         if (rippleContainer) {
           Polymer.dom(rippleContainer).appendChild(this._ripple);
+          Polymer.dom.flush();
```
Comment (notwaldorf) [inline paper-ripple-behavior.html]: Is this still needed?

File: test/index.html [hunk 2]
```diff
@@ -1,29 +1,31 @@
...
   <script src="../../web-component-tester/browser.js"></script>
 </head>
 <body>
   <script>
     WCT.loadSuites([
-      'paper-button-behavior.html',
-      'paper-radio-button-behavior.html',
-      'paper-checked-element-behavior.html',
-      'paper-ripple-behavior.html',
+      'paper-button-behavior.html?wc-shadydom=true&wc-register=true',
```
Comment (notwaldorf) [inline test/index.html]: This should be `wc-shadydom=true&wc-ce=true` now :)

**Comment by notwaldorf** [comment]
@e111077 since I reviewed this, do you want to maybe do the fixes, pretty please? (approving my own changes is weird) 🙏 🍰

File: paper-ripple-behavior.html (hunk 1 above)
Comment (e111077) [inline paper-ripple-behavior.html]: seems like a no

**Comment by e111077** [comment]
@notwaldorf fix'd!!

File: bower.json [hunk 3]
```diff
@@ -25,18 +25,36 @@
...
   "devDependencies": {
-    "iron-component-page": "polymerelements/iron-component-page#^1.0.0",
-    "iron-test-helpers": "polymerelements/iron-test-helpers#^1.0.0",
-    "paper-material": "PolymerElements/paper-material#^1.0.0",
-    "test-fixture": "PolymerElements/test-fixture#^1.0.0",
-    "web-component-tester": "^4.0.0",
-    "webcomponentsjs": "webcomponents/webcomponentsjs#^0.7.0"
+    "iron-component-page": "polymerelements/iron-component-page#2.0-preview",
+    "iron-test-helpers": "polymerelements/iron-test-helpers#2.0-preview",
+    "paper-material": "PolymerElements/paper-material#2.0-preview",
```
Comment (notwaldorf) [inline bower.json]: what do you think about not depending on paper-material and just using paper-styles/paper-material.html instead?

Comment (e111077) [inline bower.json]: ayeaye

**Comment by e111077** [comment]
removed paper-material. please look at `demo/paper-button.html` for added changes

File: bower.json [hunk 4]
```diff
@@ -25,18 +25,37 @@
...
   },
-  "ignore": []
+  "ignore": [],
+  "variants": {
+    "1.x": {
+      "dependencies": {
+        "iron-behaviors": "PolymerElements/iron-behaviors#^1.0.0",
+        "iron-checked-element-behavior": "PolymerElements/iron-checked-element-behavior#^1.0.0",
+        "paper-ripple": "PolymerElements/paper-ripple#^1.0.0",
+        "polymer": "Polymer/polymer#^1.1.0"
```
Comment (notwaldorf) [inline bower.json]: this used to be ^1.2.1

File: demo/paper-button.html [hunk 5]
```diff
@@ -9,12 +9,16 @@
 -->
 
 <link rel="import" href="../../polymer/polymer.html">
+<!-- required for 1.0 version of demo -->
 <link rel="import" href="../../paper-material/paper-material.html">
```
Comment (notwaldorf) [inline demo/paper-button.html]: `paper-styles/element-styles/paper-material` is back ported to 1.0, so this shouldn't be needed

File: demo/paper-button.html [hunk 6]
```diff
@@ -9,12 +9,16 @@
...
+<!-- required for 2.0 versopm of demo -->
+<link rel="import" href="../../paper-styles/element-styles/paper-material.html">
 <link rel="import" href="../paper-button-behavior.html">
 
 <dom-module id="paper-button">
   <template>
-    <style>
+    <!-- paper-material is included for the 2.0 version -->
```
Comment (notwaldorf) [inline demo/paper-button.html]: Comment not needed, it's the same in 1 and 2.0

File: bower.json [hunk 7]
```diff
@@ -25,18 +25,37 @@
...
+      },
+      "devDependencies": {
+        "iron-component-page": "polymerelements/iron-component-page#^1.0.0",
+        "iron-test-helpers": "polymerelements/iron-test-helpers#^1.0.0",
+        "paper-material": "PolymerElements/paper-material#^1.0.0",
```
Comment (notwaldorf) [inline bower.json]: remove paper-material, not needed anymore (we get it from paper-styles)

File: demo/paper-button.html (hunk 5 above)
Comment (e111077) [inline demo/paper-button.html]: ayeaye!

File: demo/paper-button.html (hunk 6 above)
Comment (e111077) [inline demo/paper-button.html]: i i

File: bower.json (hunk 7 above)
Comment (e111077) [inline bower.json]: 👁 👁

**Comment by e111077** [comment]
@notwaldorf j'ai fini!

**Pull request merged**